de.fu_berlin.inf.dpp.intellij/src/de/fu_berlin/inf/dpp/core/ui/eventhandler/SessionStatusChangeHandler.java

```diff
@@ -0,0 +1,118 @@
+package de.fu_berlin.inf.dpp.core.ui.eventhandler;
+
+import de.fu_berlin.inf.dpp.intellij.ui.Messages;
+import de.fu_berlin.inf.dpp.intellij.ui.util.NotificationPanel;
+import de.fu_berlin.inf.dpp.session.ISarosSession;
+import de.fu_berlin.inf.dpp.session.ISarosSessionManager;
+import de.fu_berlin.inf.dpp.session.ISessionLifecycleListener;
+import de.fu_berlin.inf.dpp.session.NullSessionLifecycleListener;
+import de.fu_berlin.inf.dpp.session.SessionEndReason;
+import de.fu_berlin.inf.dpp.session.User;
+import org.jetbrains.annotations.NotNull;
+
+import java.text.MessageFormat;
+import java.util.List;
+
+/**
+ * Notifies the local user when a session is started or the current session
+ * ended.
+ */
+public class SessionStatusChangeHandler {
+
+    public SessionStatusChangeHandler(
+        @NotNull
+            ISarosSessionManager sarosSessionManager) {
+
+        ISessionLifecycleListener lifecycleListener = new NullSessionLifecycleListener() {
+
+            @Override
+            public void sessionStarted(ISarosSession session) {
+
+                notifySessionStart(session);
+            }
+
+            @Override
+            public void sessionEnded(ISarosSession session,
+                SessionEndReason reason) {
+
+                notifySessionEnd(reason);
+            }
+        };
+
+        sarosSessionManager.addSessionLifecycleListener(lifecycleListener);
+    }
+
+    /**
+     * Notifies the user that a new session has started. If the local user is the
+     * host, this message includes a list of the current participants in the
+     * session (if present). If the local user is a client, this message
+     * includes the host whose session the user joined.
+     *
+     * @param session the session that was started
+     */
+    private void notifySessionStart(
+        @NotNull
+            ISarosSession session) {
+
+        String sessionStartMessage;
+
+        if (session.isHost()) {
+            List<User> participants = session.getRemoteUsers();
+
+            if (participants.isEmpty()) {
+                sessionStartMessage = Messages.SessionStatusChangeHandler_session_started_host_empty_message;
+
+            } else {
```

Comment:
I am not  sure if the list of users in the created session is really necessary. As far as I know, sessions can currently only be created empty (without a participant).